packages/providers/src/base.provider.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  camelCase,
+  capitalCase,
+  constantCase,
+  kebabCase,
+  pascalCase,
+  snakeCase,
+} from './utils/change-case';
+import { IOptions } from './utils/change-case/functions';
+import { deepmerge } from './utils/deepmerge.utils';
+import { Passthrough, WithPassthrough } from './utils/types';
+
+export enum CasingEnum {
+  CAMEL_CASE = 'camelCase',
+  PASCAL_CASE = 'PascalCase',
+  CAPITAL_CASE = 'CapitalCase',
+  SNAKE_CASE = 'snake_case',
+  KEBAB_CASE = 'kebabcase',
+  CONSTANT_CASE = 'CONSTANT_CASE',
+}
+
+type TransformOutput<T> = {
+  body: T;
+  headers: Record<string, string>;
+  query: Record<string, string>;
+};
+
+export abstract class BaseProvider {
+  protected casing: CasingEnum | undefined;
+
+  protected keyCaseObject: Record<string, string> = {};
+
+  protected transform<
+    T_Output = Record<string, unknown>,
+    T_Input = Record<string, unknown>,
+    T_Data = Record<string, unknown>
+  >(
+    bridgeProviderData: WithPassthrough<T_Input>,
+    triggerProviderData: T_Data
+  ): TransformOutput<T_Output> {
+    const { _passthrough = {}, ...bridgeData } = bridgeProviderData;
+    const result = this.casingTransform(bridgeData);
+
+    const fallbackPassthrough = { body: {}, headers: {}, query: {} };
+    const mergedPassthrough = deepmerge<Passthrough>(
+      fallbackPassthrough,
+      _passthrough
+    );
```

Comment:
On second thought, I think this should be renamed to `default` -

```suggestion
    const defaultPassthrough = { body: {}, headers: {}, query: {} };
    const mergedPassthrough = deepmerge<Passthrough>(
      defaultPassthrough,
      _passthrough
    );
```